ui/components/ui/box/box.js

```diff
@@ -182,56 +224,76 @@ export default function Box({
+    marginLeft &&
+      generateClassNames('box', 'margin-left', marginLeft, isValidSize),
     // Padding
-    ...generateSizeClasses(
-      'box',
-      'padding',
-      padding,
-      paddingTop,
-      paddingRight,
-      paddingBottom,
-      paddingLeft,
-    ),
-    // ---Flex/Grid alignment---
-    // if justifyContent or alignItems supplied w/o display, default to flex
-    'box--display-flex':
-      !display && (Boolean(justifyContent) || Boolean(alignItems)),
-    [`box--justify-content-${justifyContent}`]: Boolean(justifyContent),
-    [`box--align-items-${alignItems}`]: Boolean(alignItems),
-    [`box--flex-direction-${flexDirection}`]: Boolean(flexDirection),
-    [`box--flex-wrap-${flexWrap}`]: Boolean(flexWrap),
-    // text align
-    [`box--text-align-${textAlign}`]: Boolean(textAlign),
-    // display
-    [`box--display-${display}`]: Boolean(display),
-    // width & height
-    [`box--width-${width}`]: Boolean(width),
-    [`box--height-${height}`]: Boolean(height),
-    // background
-    [`box--background-color-${backgroundColor}`]: Boolean(backgroundColor),
-    ...generateSizeClasses('box', 'gap', gap),
-  });
+    padding && generateClassNames('box', 'padding', padding, isValidSize),
+    paddingTop &&
+      generateClassNames('box', 'padding-top', paddingTop, isValidSize),
+    paddingRight &&
+      generateClassNames('box', 'padding-right', paddingRight, isValidSize),
+    paddingBottom &&
+      generateClassNames('box', 'padding-bottom', paddingBottom, isValidSize),
+    paddingLeft &&
+      generateClassNames('box', 'padding-left', paddingLeft, isValidSize),
+    display && generateClassNames('box', 'display', display, isValidString),
+    gap && generateClassNames('box', 'gap', gap, isValidSize),
+    flexDirection &&
+      generateClassNames('box', 'flex-direction', flexDirection, isValidString),
+    flexWrap && generateClassNames('box', 'flex-wrap', flexWrap, isValidString),
+    justifyContent &&
+      generateClassNames(
+        'box',
+        'justify-content',
+        justifyContent,
+        isValidString,
+      ),
+    alignItems &&
+      generateClassNames('box', 'align-items', alignItems, isValidString),
+    textAlign &&
+      generateClassNames('box', 'text-align', textAlign, isValidString),
+    width && generateClassNames('box', 'width', width, isValidString),
+    height && generateClassNames('box', 'height', height, isValidString),
+    backgroundColor &&
+      generateClassNames(
+        'box',
+        'background-color',
+        backgroundColor,
+        isValidString,
+      ),
+    borderRadius &&
+      generateClassNames('box', 'rounded', borderRadius, isValidString),
+    borderStyle &&
+      generateClassNames('box', 'border-style', borderStyle, isValidString),
+    borderColor &&
+      generateClassNames('box', 'border-color', borderColor, isValidString),
+    borderWidth &&
+      generateClassNames('box', 'border-width', borderWidth, isValidSize),
```

Comment:
Changing the order of operations here to run the `generateClassNames` function only if a utility style prop exists. Previously the 'classnames' library handled the className attachment. I am concerned that we are running the `generateClassNames` for each prop instead of each category. I'm not sure if there is anyway around that though. That is probably the biggest performance concern if any at the moment?